runtime/runtime/src/config.rs

```diff
@@ -189,6 +176,44 @@ pub fn exec_fee(
         DeleteAccount(_) => cfg.delete_account_cost.exec_fee(),
     }
 }
+
+pub fn prepaid_exec_fee(
+    config: &RuntimeFeesConfig,
+    action: &Action,
+    receiver_id: &AccountId,
+    current_protocol_version: ProtocolVersion,
+) -> Gas {
+    let exec_gas = exec_fee(config, action, receiver_id, current_protocol_version);
+
+    let extra_prepaid_gas = match action {
+        Action::DeleteAccount(DeleteAccountAction { beneficiary_id }) => {
+            if checked_feature!(
+                "protocol_feature_allow_create_account_on_delete",
+                AllowCreateAccountOnDelete,
+                current_protocol_version
+            ) {
+                let sender_is_receiver = beneficiary_id == receiver_id;
+                let is_receiver_implicit =
+                    is_implicit_account_creation_enabled(current_protocol_version)
+                        && is_account_id_64_len_hex(&beneficiary_id);
+                config.action_receipt_creation_config.send_fee(sender_is_receiver)
+                    + config.action_receipt_creation_config.exec_fee()
+                    + transfer_send_fee(
+                        &config.action_creation_config,
+                        sender_is_receiver,
+                        is_receiver_implicit,
+                    )
```

Comment:
This doesn't feel right. Could you explain why we need to pay action_receipt_creation fee if the beneficiary id is not implicit?

Reply:
This is because now we create `ActionReceipt` with `Transfer` in any case (actions.rs:543) instead of refund.
If beneficiary is not implicit, we just pay for it, otherwise we also pay for implicit account creation.

Reply:
I see. Why did we decide to do this?

Reply:
I don't understand why this transfer should be free for remover, because it is the action they trigger and thus have to pay for it. I guess when the tokens might be burned it might not be fair, but now tokens are always transferred to some destination. 
Another minor opinion is that introducing a condition on account type on `action_delete_account` side will make the logic harder to understand. 
@evgenykuzyakov it was actually your change :) Is there another logic behind it?

Reply:
After thinking more about this I think the change makes sense. We should not generate a refund (which is free) to begin with.